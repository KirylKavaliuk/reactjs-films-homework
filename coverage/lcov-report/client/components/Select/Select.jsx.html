<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/Select/Select.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">client/components/Select</a> Select.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>15/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>14/14</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">21x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Link from 'components/Link/Link';
&nbsp;
import classNames from 'classnames';
&nbsp;
import Icon from 'components/Icon/Icon';
&nbsp;
import styles from './Select.scss';
&nbsp;
export default class Select extends Component {
  state = {
    open: false,
    value: 'Genre',
  };
&nbsp;
  openHandler = () =&gt; {
    this.setState({ open: true });
  }
&nbsp;
  closeHandler = () =&gt; {
    this.setState({ open: false });
  }
&nbsp;
  componentWillReceiveProps(props) {
    const { genreId } = props.match.params;
    const value = props.genres.find(_genre =&gt; _genre.id === +genreId);
&nbsp;
    this.setState({ value: value ? value.name : 'Genre' });
  }
&nbsp;
  render() {
    const { open, value } = this.state;
    const { className, genres, match } = this.props;
&nbsp;
    const selectClasses = classNames(
      styles.select,
      { [className]: className },
      { [styles.active]: match.url.includes('/genre') },
    );
&nbsp;
    const arrowClasses = classNames(styles.arrow, { [styles.arrowOpen]: open });
    const listClasses = classNames(styles.list, { [styles.listOpen]: open });
&nbsp;
    return (
      &lt;div className={ selectClasses } onMouseLeave={ this.closeHandler }&gt;
        &lt;div className={ styles.main } onClick={ this.openHandler }&gt;
          &lt;span className={ styles.value }&gt;{ value }&lt;/span&gt;
          &lt;Icon name='arrow' className={ arrowClasses }/&gt;
        &lt;/div&gt;
        &lt;div className={ listClasses }&gt;
          { genres.map(genre =&gt; (
            &lt;Link
              key={ genre.id }
              className={ styles.item }
              onClick={ this.closeHandler }
              to={ `/genre/${genre.id}` }
              clearParams={ ['query', 'movie'] }
            &gt;{ genre.name }&lt;/Link&gt;
          )) }
        &lt;/div&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
Select.defaultProps = {
  genres: [],
  match: {},
};
&nbsp;
Select.propTypes = {
  genres: PropTypes.array,
  match: PropTypes.object,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 01 2019 11:31:09 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
