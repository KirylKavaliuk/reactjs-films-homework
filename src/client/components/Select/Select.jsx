import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './Select.scss';

export default class Select extends Component {
  state = {
    open: true,
  };

  clickHandler = (event) => {
    this.setState({ open: true });
  }

  render() {
    return (
      <div
        className={ styles.select }
        onClick={ this.clickHandler }
      >
        Select
      </div>
    );
  }
}

Select.defaultProps = {

};

Select.propTypes = {

};
