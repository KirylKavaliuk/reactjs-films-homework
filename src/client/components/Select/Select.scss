@import 'styles/palette';

.select {
  background: $white;
  padding: 10px;
  box-sizing: border-box;
  border-radius: 3px;
  color: $light_blue;
  position: relative;
  width: 200px;

  &:hover {
    cursor: pointer;
  }
}

.main {
  display: flex;
  justify-content: space-between;
  padding: 5px 0;
  box-sizing: border-box;
  transition: color 0.45s ease;

  &:hover {
    color: $lighter_blue;
    fill: $lighter_blue;
  }
}

.value {
  box-sizing: border-box;
  padding: 5px 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 155px;
}

.arrow {
  width: 25px;
  height: 25px;
  fill: $light_blue;
  transition: transform 0.45s ease;
}

.item {
  width: 100%;
  border-radius: 3px;
  padding: 5px 10px;
  box-sizing: border-box;
  transition: background 0.45s ease, color 0.45s ease;
  line-height: 1.4;

  &:hover {
    background: #f1f1f1;
    color: $lighter_blue;
  }
}

.arrow-open {
  transform: rotate(180deg);
}

.list {
  position: absolute;
  width: 200px;
  box-sizing: border-box;
  background: $white;
  top: 46px;
  left: 0;
  height: 0;
  transition: all 0.45s ease;
  overflow: auto;
  z-index: -1;
  border-radius: 3px;

  &::-webkit-scrollbar {
    display: none;
  }
}

.list-open {
  padding: 10px 0 0;
  top: 46px;
  z-index: 200;
  box-sizing: border-box;
  height: 170px;
}
