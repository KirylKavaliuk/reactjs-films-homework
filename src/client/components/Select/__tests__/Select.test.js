import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import { MemoryRouter } from 'react-router-dom';

import Select from '../Select';

it('<Select/>', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <Select
        match={{
          params: { genreId: 2 },
          isExact: true,
          path: '',
          url: '',
        }}
        genres={ [
          { id: 1, name: 'genre1' },
          { id: 2, name: 'genre2' },
          { id: 3, name: 'genre3' },
        ] }
      />
    </MemoryRouter>,
  );

  const select = tree.toJSON();
  const activeValueField = select.children[0];
  expect(select).toMatchSnapshot();

  select.props.onMouseLeave();
  expect(select).toMatchSnapshot();

  activeValueField.props.onClick();
  expect(select).toMatchSnapshot();

  const sRenderer = new ShallowRenderer();
  sRenderer.render(
    <Select
      match={{
        params: { genreId: 2 },
        isExact: true,
        path: '',
        url: '',
      }}
      genres={ [
        { id: 1, name: 'genre1' },
        { id: 2, name: 'genre2' },
        { id: 3, name: 'genre3' },
      ] }
    />,
  );

  expect(sRenderer.getRenderOutput()).toMatchSnapshot();

  sRenderer.render(
    <Select
      match={{
        params: { genreId: 5 },
        isExact: true,
        path: '',
        url: '',
      }}
      genres={ [
        { id: 1, name: 'genre1' },
        { id: 2, name: 'genre2' },
        { id: 3, name: 'genre3' },
      ] }
    />,
  );

  expect(sRenderer.getRenderOutput()).toMatchSnapshot();

  sRenderer.render(
    <Select
      match={{
        params: { genreId: 2 },
        isExact: true,
        path: '',
        url: '',
      }}
      genres={ [
        { id: 1, name: 'genre1' },
        { id: 2, name: 'genre2' },
        { id: 3, name: 'genre3' },
      ] }
    />,
  );

  expect(sRenderer.getRenderOutput()).toMatchSnapshot();
});
