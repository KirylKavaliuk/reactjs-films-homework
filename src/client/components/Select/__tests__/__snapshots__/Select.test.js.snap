// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select/> 1`] = `
<div
  className="Select__select___1ZlRu"
  onMouseLeave={[Function]}
>
  <div
    className="Select__main___3EP3d"
    onClick={[Function]}
  >
    <span
      className="Select__value___2xWIB"
    >
      Genre
    </span>
    <svg
      className="Select__arrow___kPPA1"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
      />
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
    </svg>
  </div>
  <div
    className="Select__list___2Ii_p"
  >
    <a
      className="Link__link___-IU0S Select__item____FX41"
      href="/genre/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      genre1
    </a>
    <a
      className="Link__link___-IU0S Select__item____FX41"
      href="/genre/2"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      genre2
    </a>
    <a
      className="Link__link___-IU0S Select__item____FX41"
      href="/genre/3"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      genre3
    </a>
  </div>
</div>
`;

exports[`<Select/> 2`] = `
<div
  className="Select__select___1ZlRu"
  onMouseLeave={[Function]}
>
  <div
    className="Select__main___3EP3d"
    onClick={[Function]}
  >
    <span
      className="Select__value___2xWIB"
    >
      Genre
    </span>
    <svg
      className="Select__arrow___kPPA1"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
      />
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
    </svg>
  </div>
  <div
    className="Select__list___2Ii_p"
  >
    <a
      className="Link__link___-IU0S Select__item____FX41"
      href="/genre/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      genre1
    </a>
    <a
      className="Link__link___-IU0S Select__item____FX41"
      href="/genre/2"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      genre2
    </a>
    <a
      className="Link__link___-IU0S Select__item____FX41"
      href="/genre/3"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      genre3
    </a>
  </div>
</div>
`;

exports[`<Select/> 3`] = `
<div
  className="Select__select___1ZlRu"
  onMouseLeave={[Function]}
>
  <div
    className="Select__main___3EP3d"
    onClick={[Function]}
  >
    <span
      className="Select__value___2xWIB"
    >
      Genre
    </span>
    <svg
      className="Select__arrow___kPPA1"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
      />
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
    </svg>
  </div>
  <div
    className="Select__list___2Ii_p"
  >
    <a
      className="Link__link___-IU0S Select__item____FX41"
      href="/genre/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      genre1
    </a>
    <a
      className="Link__link___-IU0S Select__item____FX41"
      href="/genre/2"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      genre2
    </a>
    <a
      className="Link__link___-IU0S Select__item____FX41"
      href="/genre/3"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      genre3
    </a>
  </div>
</div>
`;
